### PR TITLE
[Events Orderbook] Implement event polling for state updates. (5)

### PR DESCRIPTION
**Pull request by nlordell**
This PR implements initial logic to periodically poll the node for new events and update the account state accordingly.

### Test Plan

Plan on implementing an integrity test to ensure that the retrieved orderbook matches the onchain filtered orderbook.

### CODE REVIEW

**Review by e00E — APPROVED** [review]

File: src/orderbook/index.ts [hunk 1]
```diff
@@ -129,15 +180,74 @@ export class StreamedOrderbook {
         fromBlock + this.options.blockPageSize - 1,
         endBlock,
       )
-      console.error(`fetching page ${fromBlock}-${toBlock}`)
 
+      console.error(`fetching page ${fromBlock}-${toBlock}`)
```
Comment (fleupold) [inline src/orderbook/index.ts:184]: just noticed that these (and some other new non error messages in this PR) are printed to error. Is there a reason for that?

File: src/orderbook/index.ts [hunk 2]
```diff
@@ -98,17 +121,45 @@ export class StreamedOrderbook {
...
+
     return orderbook
   }
 
   /**
-   * Dumps the current account state to console.
+   * Stops the orderbook so that it no longer performs any updates.
+   *
+   * @throws
+   * Throws any error that may have occurred when updating and applying new
```
Comment (fleupold) [inline src/orderbook/index.ts:137]: I find it a bit weird that this method can potentially throw an error that has happened long in the past and when we are about to end polling anyways. Is there a reason for it? 
Should we maybe just return it or have a bit more information about when and how often this happened?

**Review by fleupold — APPROVED** [review]
It's a shame we will be lagging ~90 seconds behind the head off the chain. Do you plan to make it so that we can " "optimistically" apply the events until latest as we do in the services?

File: src/orderbook/index.ts (hunk 1 above)
Comment (nlordell) [reply src/orderbook/index.ts:184]: Just to print to stderr - these will be cleaned up.

File: src/orderbook/index.ts (hunk 2 above)
Comment (nlordell) [reply src/orderbook/index.ts:137]: The idea here is that we have a polling fiber but no way to check whether or not it error-ed. Eventually, we can:
- Emit "error" events when an error is encountered.
- Also error every time the orderbook is constructed/retrieved from the current state if the fiber has errored.

**Comment by nlordell** [comment]
> Do you plan to make it so that we can "optimistically" apply the events until latest as we do in the services?

Yes, the idea is when the orderbook is retrieved it will be constructed from the current state + the pending events.

**Comment by nlordell** [comment]
This PR has been moved to gnosis/dex-contracts#724 and gnosis/dex-contracts#730

**Pull request closed without merging**